snuba/pipeline/query_pipeline.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+import sentry_sdk
+
+from snuba.datasets.plans.query_plan import ClickhouseQueryPlan, QueryRunner
+from snuba.request import Request
+from snuba.web import QueryResult
+
+
+class QueryPipeline(ABC):
+    """
+    Contains the instructions to build the query plans for the requested query
+    (this can be one or multiple query plans). The QueryPipeline runs the query
+    plan processors, executes the query plans and returns the relevant result.
```

Comment:
Adding an example of when there can be multiple query plan to execute may be useful since this is not an intuitive case.